libs/server/Custom/CustomCommandManagerSession.cs

```diff
@@ -12,25 +12,40 @@ internal sealed class CustomCommandManagerSession
     {
         readonly CustomCommandManager customCommandManager;
         public readonly (CustomTransactionProcedure, int)[] sessionTransactionProcMap;
+        public readonly CustomProcedure[] sessionCustomProcMap;
+
 
         public CustomCommandManagerSession(CustomCommandManager customCommandManager)
         {
             this.customCommandManager = customCommandManager;
             sessionTransactionProcMap = new (CustomTransactionProcedure, int)[CustomCommandManager.MaxRegistrations];
+            sessionCustomProcMap = new CustomProcedure[CustomCommandManager.MaxRegistrations];
+        }
+
+        public CustomProcedure GetCustomProcedure(int id, RespServerSession respServerSession)
+        {
+            if (sessionCustomProcMap[id] == null)
```

Comment:
Should we add a bounds check for the id? Also, are we sure we want to throw if the id is not found? Maybe just return null?

Reply:
This id is from matching the commands in the global arrays in CustomCommandManager so if the id isn't found that should be an error condition.